src/needless_continue.rs

```diff
@@ -0,0 +1,547 @@
+    ret.push_str(&align_snippets(&[&if_code, "\n// Merged code follows...", &to_annex]));
+    ret 
+}
+
+/// Constructs a suggestion for when there is just a continue statement in an
+/// if block, with the meat of the code being in its else arm and possibly
+/// after it. Concretely for this general case,
+///     
+///     loop {
+///         // region A
+///         if cond {
+///             continue;
+///             // potentially more code here.
+///         } else {
+///             // region B
+///         }
+///         // region C
+///     }
+///
+/// the suggestion returned is
+///
+///     loop {
+///         // region A
+///         if !cond {
+///             // region B
+///             // region C
+///         }
+///     }
+///
+/// Note that the if condition is negated, and this function tries to simplify
+/// the negated expression as much as possible, to cover cases like
+/// !(e1 < e2) -> e1 >= e2 and !(E1 || E2) -> !E1 && !E2 etc.
+///
+fn suggestion_snippet_for_continue_inside_if(ctx: &LateContext,
+                                             if_cond: &Expr,
+                                             else_expr: &Expr,
+                                             header: &str) -> String {
+
+    let cond_code = &pprust::expr_to_string(&simplify(&negate_expr(if_cond)));
+    let if_code = &format!("if {} {{", cond_code);
+
+    let span = Span { lo: else_expr.span.lo + BytePos::from_usize(1usize),
+                      hi: else_expr.span.hi,
+                      expn_id: else_expr.span.expn_id };
+    let else_code = &trim_indent(&snippet(ctx, span, "..").into_owned(), false);
+
+    // Since we do not really know how much to indent here, we'll just indent
+    // the if block with 4 spaces.
+    let mut ret = String::from(header);
+    ret.push_str(if_code);
+    ret.push_str(&left_pad_with_spaces(&trim_indent(else_code, false), 4usize));
+    ret
+}
+
+/// Created a Spanned<T> with a dummy Span.
+fn mk_spanned<T>(node: T) -> Spanned<T> {
+    Spanned {
+        node: node,
+        span: dummy_span()
+    }
+}
+
+/// Creates a dummy Span with all fields set to zero.
+fn dummy_span() -> Span {
+    Span { lo: BytePos::from_usize(0usize),
+           hi: BytePos::from_usize(0usize),
+           expn_id: ExpnId::from_u32(0u32) }
+}
+
+/// Returns the dual comparison operator (under boolean negation) of `op`.
+fn dual_comparison_operator(op: BinOp_) -> Option<BinOp_> {
+    match op {
+        BiLt  => Some(BiGe),
+        BiLe  => Some(BiGt),
+        BiGt  => Some(BiLe),
+        BiGe  => Some(BiLt),
+        BiEq  => Some(BiNe),
+        BiNe  => Some(BiEq),
+        _     => None,
+    }
+}
+
+/// Returns the de Morgan's dual operator of `op` if it is either `&&` or `||`.
+fn dual_boolean_operator(op: BinOp_) -> Option<BinOp_> {
+    match op {
+        BiAnd => Some(BiOr),
+        BiOr  => Some(BiAnd),
+        _     => None,
+    }
+}
+
+fn is_comparison_operator(op: BinOp_) -> bool {
```

Comment:
I think there is an `is_comparison_binop(_)` function in `librustc_front::util`.